src/Core/Framework/App/Subscriber/SystemLanguageChangedSubscriber.php

```diff
@@ -0,0 +1,132 @@
+    {
+        return [
+            SystemLanguageChangeEvent::class => 'onSystemLanguageChanged',
+        ];
+    }
+
+    public function onSystemLanguageChanged(SystemLanguageChangeEvent $event): void
+    {
+        if ($this->snippetRepository === null) {
+            return;
+        }
+
+        $context = Context::createDefaultContext();
+
+        $snippets = $this->getAppAdministrationSnippets($this->snippetRepository, $context);
+        if ($snippets->count() === 0) {
+            return;
+        }
+
+        $appIds = $this->getApps($context)->getIds();
+
+        $appsWithSnippets = [];
+        foreach ($snippets as $snippet) {
+            if (\in_array($snippet->getAppId(), $appIds, true) && !\in_array($snippet->getAppId(), $appsWithSnippets, true)) {
+                $appsWithSnippets[] = $snippet->getAppId();
+            }
+        }
+
+        $previousLocale = $this->getPreviousLocale($event->previousLanguageId, $context);
+        $newLocale = $this->getNewLocale($context);
+
+        foreach ($appsWithSnippets as $appId) {
+            $snippetToClone = $snippets->filter(fn (AppAdministrationSnippetEntity $snippet) => $appId === $snippet->getAppId() && $snippet->getLocaleId() === $newLocale->getId())->first();
+            \assert($snippetToClone instanceof AppAdministrationSnippetEntity);
+
+            $snippetWithPreviousLocaleExists = $snippets->filter(fn (AppAdministrationSnippetEntity $snippet) => $appId === $snippet->getAppId() && $snippet->getLocaleId() === $previousLocale->getId())->first();
+            if ($snippetWithPreviousLocaleExists) {
+                continue;
+            }
+
+            $this->snippetRepository->create([[
+                'appId' => $snippetToClone->getAppId(),
+                'localeId' => $previousLocale->getId(),
+                'value' => $snippetToClone->getValue(),
+            ]], $context);
+        }
+    }
+
+    /**
+     * Returns the locale of the previous default language.
+     */
+    private function getPreviousLocale(string $languageId, Context $context): LocaleEntity
+    {
+        $criteria = new Criteria([$languageId]);
+        $criteria->addAssociation('locale');
+
+        $language = $this->languageRepository->search($criteria, $context)->first();
+        \assert($language instanceof LanguageEntity);
+
+        $locale = $language->getLocale();
+        \assert($locale instanceof LocaleEntity);
+
+        return $locale;
+    }
+
+    /**
+     * Returns the locale of the new default language.
+     */
+    private function getNewLocale(Context $context): LocaleEntity
+    {
+        $criteria = new Criteria([Defaults::LANGUAGE_SYSTEM]);
+        $criteria->addAssociation('locale');
+
+        $language = $this->languageRepository->search($criteria, $context)->first();
+        \assert($language instanceof LanguageEntity);
+
+        $locale = $language->getLocale();
+        \assert($locale instanceof LocaleEntity);
+
+        return $locale;
+    }
```

Comment:
They are basically the same :sweat_smile: 

```suggestion
    private function getLocale(string $languageId, Context $context): LocaleEntity
    {
        $criteria = new Criteria([$languageId]);
        $criteria->addAssociation('locale');

        $language = $this->languageRepository->search($criteria, $context)->first();
        \assert($language instanceof LanguageEntity);

        $locale = $language->getLocale();
        \assert($locale instanceof LocaleEntity);

        return $locale;
    }

    /**
     * Returns the locale of the new default language.
     */
    private function getDefaultLocale(Context $context): LocaleEntity
    {
        return $this->getLocale(Defaults::LANGUAGE_SYSTEM);
    }
```